internal/provider/vxc_resource.go

```diff
@@ -0,0 +1,1494 @@
+	cspConnection
+	ConnectType       types.String                                `tfsdk:"connect_type"`
+	ResourceName      types.String                                `tfsdk:"resource_name"`
+	ResourceType      types.String                                `tfsdk:"resource_type"`
+	VLAN              types.Int64                                 `tfsdk:"vlan"`
+	Interfaces        []*cspConnectionVirtualRouterInterfaceModel `tfsdk:"interfaces"`
+	IPAddresses       []types.String                              `tfsdk:"ip_addresses"`
+	VirtualRouterName types.String                                `tfsdk:"virtual_router_name"`
+}
+
+// cspConnectionVirtualRouterInterfaceModel represents the configuration of a CSP connection for Virtual Router interface.
+type cspConnectionVirtualRouterInterfaceModel struct {
+	IPAddresses []types.String `tfsdk:"ip_addresses"`
+}
+
+// cspConnectionTransit represents the configuration of a CSP connection for a Transit VXC.
+type cspConnectionTransit struct {
+	cspConnection
+	ConnectType        types.String `tfsdk:"connect_type"`
+	ResourceName       types.String `tfsdk:"resource_name"`
+	ResourceType       types.String `tfsdk:"resource_type"`
+	CustomerIP4Address types.String `tfsdk:"customer_ip4_address"`
+	CustomerIP6Network types.String `tfsdk:"customer_ip6_network"`
+	IPv4GatewayAddress types.String `tfsdk:"ipv4_gateway_address"`
+	IPv6GatewayAddress types.String `tfsdk:"ipv6_gateway_address"`
+}
+
+// virtualRouterModel maps the virtual router schema data.
+type virtualRouterModel struct {
+	MCRAsn             types.Int64  `tfsdk:"mcr_asn"`
+	ResourceName       types.String `tfsdk:"resource_name"`
+	ResourceType       types.String `tfsdk:"resource_type"`
+	Speed              types.Int64  `tfsdk:"speed"`
+	BGPShutdownDefault types.Bool   `tfsdk:"bgp_shutdown_default"`
+}
+
+// vllConfigModel maps the VLL configuration schema data.
+type vllConfigModel struct {
+	AEndVLAN      types.Int64  `tfsdk:"a_vlan"`
+	BEndVLAN      types.Int64  `tfsdk:"b_vlan"`
+	Description   types.String `tfsdk:"description"`
+	ID            types.Int64  `tfsdk:"id"`
+	Name          types.String `tfsdk:"name"`
+	RateLimitMBPS types.Int64  `tfsdk:"rate_limit_mbps"`
+	ResourceName  types.String `tfsdk:"resource_name"`
+	ResourceType  types.String `tfsdk:"resource_type"`
+}
+
+// vxcApprovalModel maps the approval schema data.
+type vxcApprovalModel struct {
+	Status   types.String `tfsdk:"status"`
+	Message  types.String `tfsdk:"message"`
+	UID      types.String `tfsdk:"uid"`
+	Type     types.String `tfsdk:"type"`
+	NewSpeed types.Int64  `tfsdk:"new_speed"`
+}
+
+// vxcOrderEndPointConfigurationModel maps the end point configuration schema data.
+type vxcOrderEndPointConfigurationModel struct {
+	ProductUID              types.String             `tfsdk:"product_uid,omitempty"`
+	VLAN                    types.Int64              `tfsdk:"vlan,omitempty"`
+	PartnerConfig           *vxcPartnerConfiguration `tfsdk:"partner_config,omitempty"`
```

Comment:
This should probably be modeled where each partner is it's own field rather than all the fields being set in the schema at the root level. Could also be a general purpose config section where it has options like:

```go
type vxcOrderEndPointConfigurationModel struct {
	ProductUID              types.String             `tfsdk:"product_uid,omitempty"`
	VLAN                    types.Int64              `tfsdk:"vlan,omitempty"`
	Config *endConfig
}

type endConfig struct {
	MVEConfig vxcOrderMVEConfigModel
	AWSConfig vxcPartnerConfigAWSModel
	// ... etc
}
```
Then we can enforce that only one is set at a time using terraform validation rules.